### PR TITLE
Fix links to makefile and gemfile for website

### LINKED ISSUE #3524: Broken Links with Make file and and Gemfile  on site_contribution.md
https://github.com/projectcontour/contour/blob/main/SITE_CONTRIBUTION.md#local-site-testing
I am making a PR to fix this

### PR DESCRIPTION
**Pull request by prondubuisi**
Fixes #3524
Signed-off-by: prondubuisi <onyemenamndu@gmail.com>

### CODE REVIEW

File: SITE_CONTRIBUTION.md [hunk 1]
```diff
@@ -109,10 +109,10 @@ Perform local tests before creating a PR to detect issues such as broken links.
 
 **Using docker:**
 
-The easiest way to run the site locally is by using the top-level [Makefile](../Makefile). This has a `site-devel` target and runs a [Jekyll][6] development server in Docker, which is useful for checking local documentation changes.
+The easiest way to run the site locally is by using the top-level [Makefile](/Makefile). This has a `site-devel` target and runs a [Jekyll][6] development server in Docker, which is useful for checking local documentation changes.
```
Comment (sunjayBhatia) [inline SITE_CONTRIBUTION.md:112]:
```suggestion
The easiest way to run the site locally is by using the top-level [Makefile](./Makefile). This has a `site-devel` target and runs a [Jekyll][6] development server in Docker, which is useful for checking local documentation changes.
```

File: SITE_CONTRIBUTION.md [hunk 2]
```diff
@@ -109,10 +109,10 @@ Perform local tests before creating a PR to detect issues such as broken links.
...
 By default, the site will be available on [port 4000](http://127.0.0.1:4000/).
 
-Note that the version of [Jekyll][6] used by the site is pinned in the [Gemfile](./Gemfile.lock). The `site-devel` target uses the matching versioned Docker image to ensure compatibility.
+Note that the version of [Jekyll][6] used by the site is pinned in the [Gemfile](/site/Gemfile.lock). The `site-devel` target uses the matching versioned Docker image to ensure compatibility.
```
Comment (sunjayBhatia) [inline SITE_CONTRIBUTION.md:115]:
```suggestion
Note that the version of [Jekyll][6] used by the site is pinned in the [Gemfile](./site/Gemfile.lock). The `site-devel` target uses the matching versioned Docker image to ensure compatibility.
```

**Review by sunjayBhatia — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a small nit to use a fully relative reference

File: SITE_CONTRIBUTION.md (hunk 2 above)
Comment (prondubuisi) [reply SITE_CONTRIBUTION.md:115]: Seen @sunjayBhatia , making the updates.

File: SITE_CONTRIBUTION.md (hunk 1 above)
Comment (prondubuisi) [reply SITE_CONTRIBUTION.md]: Fixed

File: SITE_CONTRIBUTION.md (hunk 2 above)
Comment (prondubuisi) [reply SITE_CONTRIBUTION.md]: Fixed!

**Review by skriss — APPROVED** [review]

**Pull request merged**